File: python/p010.py
#!/usr/bin/python


def sum_primes_below_n(n):
    primes = [True] * n
    primes[0], primes[1] = [None] * 2
    prod = 0
    for ind, val in enumerate(primes):
        if val is True:
            primes[ind * 2::ind] = [False] * (((n - 1) // ind) - 1)
            prod += ind
    return prod


print(sum_primes_below_n(2000000))

File: rust/src/bin/p010.rs

fn sum_primes_below_n(n: usize) -> usize {
    let mut primes = vec![true; n];
    primes[0] = false;
    primes[1] = false;
    let mut ind = 0;
    let mut prod = 0;
    while ind < primes.len() {
        if primes[ind] == true {
            let mut i = ind * ind;
            while i < primes.len() {
                primes[i] = false;
                i += ind;
            }
            prod += ind;
        }
        ind += 1;
    }
    prod
}

fn main() {
    println!("{}", sum_primes_below_n(2000000));
}
